src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4001,17 +4001,17 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers)
             ExplicitInterfaceSpecifierSyntax explicitInterfaceOpt;
             SyntaxToken identifierOrThisOpt;
             TypeParameterListSyntax typeParameterList;
-
-            this.ParseMemberName(out explicitInterfaceOpt, out identifierOrThisOpt, out typeParameterList, isEvent: true);
+            bool aggressiveErrorRecoverySuppressed;
+            this.ParseMemberName(out explicitInterfaceOpt, out identifierOrThisOpt, out typeParameterList, isEvent: true, out aggressiveErrorRecoverySuppressed);
 
             // If we got an explicitInterfaceOpt but not an identifier, then we're in the special
             // case for ERR_ExplicitEventFieldImpl (see ParseMemberName for details).
-            if (explicitInterfaceOpt != null && identifierOrThisOpt == null)
+            if (explicitInterfaceOpt != null && identifierOrThisOpt == null || aggressiveErrorRecoverySuppressed)
```

Comment:
>explicitInterfaceOpt != null && identifierOrThisOpt == null || aggressiveErrorRecoverySuppressed [](start = 16, length = 96)

It looks like this condition can safely be replaced with ```explicitInterfaceOpt != null && this.CurrentToken.Kind != SyntaxKind.OpenBraceToken``` and the extra parameter for ```ParseMemberName``` is not necessary.